frontend/src/components/AuthProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { User } from "../model";
+import { SNOWPACK_PUBLIC_API_URL } from "../utils";
+import { Buffer } from "buffer";
+import { useNavigate } from "react-router-dom";
+
+export const AuthContext = React.createContext(null);
+
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = React.useState();
+  const navigate = useNavigate();
+
+  React.useEffect(() => {
+    let user = window.localStorage.getItem("user");
+    if (user) user = JSON.parse(user);
+    setUser(user);
+  }, []);
+
+  React.useEffect(() => {
+    if (user) window.localStorage.setItem("user", JSON.stringify(user));
+  }, [user]);
```

Comment:
You might need cleanup functions here? 
https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1

Reply:
Thanks for the input Henrike. I added a commit to fix the error. The logout frontend logic should be out of the async call I think

Reply:
This error happens because of the redirect to login. We need to cancel the refresh of the page when redirecting, which I am doing in my last commit. Still sometimes I see the error happening.